docs/pages/admin-guides/access-controls/rbac/getting-started-rbac.mdx

```diff
@@ -0,0 +1,315 @@
+
+External users authenticate using external Identity Providers (IdP), such as Okta, Google, GitHub, Microsoft Entra ID, or other SAML/OIDC providers.
+These users do not manage their passwords or credentials within Teleport itself. Instead, they authenticate using their external SSO credentials.
+
+Instead of manually assigning roles in Teleport, external users’ roles are mapped based on attributes from the identity provider (e.g., groups in Okta or teams in GitHub). 
+For example, a user in the "admin" group in Okta can be mapped to the admin role in Teleport. External users are useful for larger organizations or those that want to centralize 
+identity management via an existing SSO provider. This is often preferred for enterprise environments where identity management is handled outside Teleport.
+
+### Bots
+
+Machine ID provides machines with an identity that can authenticate to the Teleport cluster. This identity is known as a bot. Bots are represented in Teleport by a user and a role resource and can 
+be created via the tctl bots add command. Unlike human users, who use a password, MFA, or SSO, bot users join the cluster as Teleport services using a join method. 
+They can still join even if local auth is disabled.
+
+See the [Machine ID introduction](../../../enroll-resources/machine-id/introduction.mdx for more information.
+
+## Teleport roles
+
+Roles in Teleport define the permissions and access controls for users within a Teleport cluster. They determine what resources (like SSH servers, databases, Kubernetes clusters, web apps, and desktops) 
+a user can access and what actions they can perform on those resources.
+
+Roles control access to specific resources and functions, such as viewing logs, editing configurations, or accessing specific servers, 
+and are used for internal users (created directly in Teleport) and external users (authenticated via Single Sign-On, like Okta or GitHub).
+Users can be assigned one or more roles, and the permissions of multiple roles combine to form a user’s overall capabilities.
+
+Admins create and assign roles to manage who can access which parts of the infrastructure.
+
+Some examples of roles include:
+
+- **access**: Grants basic access to resources.
+- **editor**: Allows users to modify configuration settings.
+- **auditor**: Allows users to view logs and auditing activities.
+
+Roles enforce Role-Based Access Control (RBAC), ensuring that users only have the permissions needed for their specific tasks. 
+See [Teleport Access Controls Reference](../../../reference/access-controls/roles.mdx) for in-depth information about types of roles, options, and usage.
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+
+(!docs/pages/includes/permission-warning.mdx!)
+
+## Step 1/3. Add local users with preset roles
```

Comment:
The steps—which are the same as the current guide's—cover a lot of ground quickly, which is what [this issue](https://github.com/gravitational/teleport/issues/39241) points out. 

Is there a way to simplify the guide to walk a user through Teleport's resource tagging and matching logic so a user understands how roles work? Otherwise, the guide leaves a user to pick a preset role, then presents a complex YAML document and assumes the user can figure it out.

Reply:
Introduced and expanded on these concepts, added definitions and YAML explanation